event/src/main/scala/com/improving/app/event/domain/Event.scala

```diff
@@ -0,0 +1,605 @@
+                Some(newMeta)
+              )
+            )
+          )
+        } else
+          Left(StateError(validationErrorsOpt.get.message))
+      case Right(info: EventInfo) =>
+        Right(
+          EventEventResponse(
+            EventScheduled(
+              scheduleEventCommand.eventId,
+              Some(info),
+              Some(newMeta)
+            )
+          )
+        )
+    }
+  }
+
+  private def rescheduleEvent(
+      info: EventInfo,
+      meta: EventMetaInfo,
+      rescheduleEventCommand: RescheduleEvent
+  ): Either[Error, EventResponse] = {
+    val now = Some(Timestamp(Instant.now(clock)))
+    val newMeta = meta.copy(
+      lastModifiedOn = now,
+      lastModifiedBy = rescheduleEventCommand.onBehalfOf,
+      scheduledOn = now,
+      scheduledBy = rescheduleEventCommand.onBehalfOf,
+      currentState = EVENT_STATE_SCHEDULED,
+      eventStateInfo = Some(EventStateInfo(EventStateInfo.Value.ScheduledEventInfo(ScheduledEventInfo())))
+    )
+
+    Right(
+      EventEventResponse(
+        EventScheduled(
+          rescheduleEventCommand.eventId,
+          Some(info.copy(expectedStart = rescheduleEventCommand.start, expectedEnd = rescheduleEventCommand.end)),
+          Some(newMeta)
+        )
+      )
+    )
+  }
+
+  private def startEvent(
+      info: EventInfo,
+      meta: EventMetaInfo,
+      startEventCommand: StartEvent
+  ): Either[Error, EventResponse] = {
+    val now = Some(Timestamp(Instant.now(clock)))
+    val newMeta = meta.copy(
+      lastModifiedOn = now,
+      lastModifiedBy = startEventCommand.onBehalfOf,
+      scheduledOn = now,
+      scheduledBy = startEventCommand.onBehalfOf,
+      currentState = EVENT_STATE_INPROGRESS,
+      eventStateInfo = Some(EventStateInfo(EventStateInfo.Value.InProgressEventInfo(InProgressEventInfo(now)))),
+      actualStart = now
+    )
+
+    Right(
+      EventEventResponse(
+        EventScheduled(
+          startEventCommand.eventId,
+          Some(info),
+          Some(newMeta)
+        )
+      )
+    )
+  }
+
+  private def delayEvent(
+      info: Either[EditableEventInfo, EventInfo],
+      meta: EventMetaInfo,
+      delayEventCommand: DelayEvent
+  ): Either[Error, EventResponse] = {
+    val now = Some(Timestamp(Instant.now(clock)))
+    val newMeta = meta.copy(
+      lastModifiedOn = now,
+      lastModifiedBy = delayEventCommand.onBehalfOf,
+      scheduledOn = now,
+      scheduledBy = delayEventCommand.onBehalfOf,
+      currentState = EVENT_STATE_DELAYED,
+      eventStateInfo = Some(
+        EventStateInfo(
+          EventStateInfo.Value.DelayedEventInfo(
+            DelayedEventInfo(reason = delayEventCommand.reason, timeStartedOpt = meta.actualStart)
+          )
+        )
+      )
+    )
+    info match {
+      case Left(e: EditableEventInfo) =>
+        val validationErrorsOpt = draftTransitionEventInfoValidator(e)
+        if (validationErrorsOpt.isEmpty) {
+          Right(
+            EventEventResponse(
```

Comment:
EventEventResponse :\ I wonder if we would do well to find an alternate name for Event to avoid collision with actor model internal nomenclature.

Reply:
https://www.thesaurus.com/browse/event I think "Affair" is the closest in meaning but "Event" is much more idiomatic in the domain, I think. Oh well.